hbase-client/src/main/java/org/apache/hadoop/hbase/ipc/IPCUtil.java

```diff
@@ -159,8 +163,19 @@ static RemoteException createRemoteException(final ExceptionResponse e) {
   }
 
   /** Returns True if the exception is a fatal connection exception. */
-  static boolean isFatalConnectionException(final ExceptionResponse e) {
-    return e.getExceptionClassName().equals(FatalConnectionException.class.getName());
+  static boolean isFatalConnectionException(ExceptionResponse e) {
+    if (e.getExceptionClassName().equals(FatalConnectionException.class.getName())) {
+      return true;
+    }
+    // try our best to check for sub classes of FatalConnectionException
+    try {
+      return e.getExceptionClassName() != null && FatalConnectionException.class
+        .isAssignableFrom(Class.forName(e.getExceptionClassName()));
```

Comment:
Might be good to pass initialize=false to Class.forName

Reply:
Good point. Let me check if this can work.

Reply:
Seems worked. I also add a UT for the method.